src/core/border_router/routing_manager.cpp

```diff
@@ -330,6 +401,36 @@ void RoutingManager::HandleReceived(const InfraIf::Icmp6Packet &aPacket, const I
     return;
 }
 
+Error RoutingManager::HandleReceivedHostAddress(const char *        aHostName,
+                                                const Ip6::Address *aAddresses,
+                                                uint8_t             aNumAddresses)
+{
+    Error       error = kErrorNotImplemented;
+    Ip6::Prefix prefix;
+
+    OT_UNUSED_VARIABLE(aAddresses);
+    OT_UNUSED_VARIABLE(aNumAddresses);
+    OT_UNUSED_VARIABLE(prefix);
+
+    VerifyOrExit(strcmp(aHostName, kWellKnownIpv4OnlyName) == 0);
+
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_NAT64_ENABLE
+    prefix.Clear();
+    // TODO: use OT_IP4_ADDRESS_SIZE when it's ready
+    if (aAddresses != nullptr && aNumAddresses > 0 &&
+        (memcmp(&aAddresses->mFields.m8[OT_IP6_ADDRESS_SIZE - 4], kWellKnownIpv4OnlyAddress1, 4) == 0 ||
```

Comment:
nit: should avoid hardcoding `4`.

Reply:
Yes, it will be updated with OT_IP4_ADDRESS_SIZE once [PR#7824](https://github.com/openthread/openthread/pull/7824) is merged.